src/core/server/saved_objects/migrations/core/document_migrator.test.ts

```diff
@@ -165,13 +165,13 @@ describe('DocumentMigrator', () => {
       attributes: { name: 'Tyler' },
       acl: 'anyone',
       migrationVersion: {},
-    });
+    } as SavedObjectUnsanitizedDoc);
```

Comment:
our types are stricter than the code...  although the code still supports plugins adding their own root properties there's no plugins using this and we should probably remove this functionality.

Reply:
Yea, I feel like migrations are more permissive than they should, at least for plugins.